libV2/schemaUtils.js

```diff
@@ -1914,7 +1997,33 @@ let QUERYPARAM = 'query',
         return;
       }
 
-      let paramValue = resolveValueOfParameter(context, param);
+      let propertyDetails = {},
+        properties = {},
+        keyName,
+        paramValue = resolveValueOfParameter(context, param);
+
+      if (param && param.schema) {
+        const { name, schema } = param;
+        keyName = name;
+        properties = {
+          type: schema.type || 'unknown',
+          format: schema.format || undefined,
+          default: schema.default || undefined,
+          required: param.required || false,
+          deprecated: param.deprecated || false,
+          enum: schema.enum || undefined,
+          minLength: schema.minLength || undefined,
+          maxLength: schema.maxLength || undefined,
+          minimum: schema.minimum || undefined,
+          maximum: schema.maximum || undefined,
+          pattern: schema.pattern || undefined,
+          example: schema.example || undefined
+        };
+      }
+      propertyDetails = { keyName, properties };
+      if (keyName && param.schema && param.schema.type) {
+        queryParamTypes.push(propertyDetails);
```

Comment:
P1: `propertyDetails` can be renamed to `queryParamTypeInfo` to make it consistent with the array name `queryParamTypes`

Same for other places where you have this same piece of code